File: Famouspersonbiographies/app/src/main/java/np/com/sarojb/famouspersonbiographies/MainActivity.java
package np.com.sarojb.famouspersonbiographies;

import android.content.ContentValues;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import com.google.android.material.snackbar.Snackbar;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    DatabaseHelper databaseHelper;
    private Button button_allpeople;
    private Button button_favourite;
    private Button button_requestmore;
    private ImageView imageview_info;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        databaseHelper = new DatabaseHelper(this);
        ArrayList<FamousPersons> arrayList = new ArrayList<>();
        arrayList = databaseHelper.getFamousPersonFromDatabase();
        final int a=  arrayList.size();

        RequestQueue requestQueue;
        requestQueue = Volley.newRequestQueue(this);
        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.GET, "http://www.sarojb.com.np/jsondata.json", null,
                new Response.Listener<JSONObject>() {
                    @Override
                    public void onResponse(JSONObject response) {
                        try {
                            JSONArray jsonArray = response.getJSONArray("allpersons");
                            int b = a;
                            for (int i = 0; i < jsonArray.length(); i++) {
                                ContentValues contentValues = new ContentValues();
                                JSONObject jsonOb = jsonArray.getJSONObject(i);
                                String firstname = jsonOb.getString("name");
                                String imageUrl = jsonOb.getString("imageUrl");
                                String fieldOfWork = jsonOb.getString("fieldOfWork");
                                String shortdesc = jsonOb.getString("shortdesc");
                                String longdesc = jsonOb.getString("longdesc");
                                try{
                                    contentValues.put("id",b+1);
                                    contentValues.put("name", firstname);
                                    contentValues.put("imageUrl", imageUrl);
                                    contentValues.put("fieldOfWork", fieldOfWork);
                                    contentValues.put("shortdesc", shortdesc);
                                    contentValues.put("longdesc", longdesc);
                                    databaseHelper.insertData(contentValues);
                                    b += 1;
                                }
                                catch (Exception e){
                                    e.printStackTrace();
                                }

                            }
                            Log.e("onResponse: ", "IN response" + response.getString(""));
                        } catch (JSONException e) {
                            Log.e("Error", "error");
                        }
                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Log.e("Error", "onErrorResponse: ");
            }
        });
        requestQueue.add(jsonObjectRequest);

//        databaseHelper = new DatabaseHelper(this);
//        ContentValues contentValues = new ContentValues();
//        contentValues.put("name", "Albert Bahadur");
//
//        databaseHelper.insertData(contentValues);

        initViews();
        button_allpeople.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, DetailActivity.class);
                startActivity(intent);
            }
        });
        button_favourite.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Snackbar.make(v, "OOPS Still working on this", Snackbar.LENGTH_LONG).show();
            }
        });

        button_requestmore.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Snackbar.make(v, "OOPS Still working on this", Snackbar.LENGTH_LONG).show();
            }
        });
        imageview_info.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "working on this", Toast.LENGTH_SHORT).show();
            }
        });
        button_favourite.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, FavouriteActivity.class);
                startActivity(intent);
            }
        });

    }

    private void initViews() {
        button_allpeople = findViewById(R.id.button_allpeople);
        button_favourite = findViewById(R.id.button_favourite);
        button_requestmore = findViewById(R.id.button_requestmore);
        imageview_info = findViewById(R.id.imageview_info);
    }
}

File: tictactoe/app/src/main/java/np/com/sarojb/tictactoe/MainActivity.kt
package np.com.sarojb.tictactoe

import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun click_button(view: View) {
        val selected_button: Button = view as Button
        var button_id = 0;
        when (selected_button.id) {
            R.id.button_1 -> button_id = 1
            R.id.button_2 -> button_id = 2
            R.id.button_3 -> button_id = 3
            R.id.button_4 -> button_id = 4
            R.id.button_5 -> button_id = 5
            R.id.button_6 -> button_id = 6
            R.id.button_7 -> button_id = 7
            R.id.button_8 -> button_id = 8
            R.id.button_9 -> button_id = 9
        }
//        Toast.makeText(this, "id" + button_id, Toast.LENGTH_LONG).show()
        playGame(button_id, selected_button)

    }

    var player1 = ArrayList<Int>()
    var player2 = ArrayList<Int>()
    var active_player = 1

    private fun playGame(buttonId: Int, selectedButton: Button) {
        if (active_player == 1) {
            player1.add(buttonId)
            selectedButton.text = "X"
            selectedButton.setBackgroundColor(Color.GREEN)
            active_player = 2
        } else {
            player2.add(buttonId)
            selectedButton.text = "0"
            selectedButton.setBackgroundColor(Color.BLUE)
            active_player = 1
        }
        selectedButton.isEnabled = false
        winner()
    }

    fun winner() {
        var winner: Int = -1
        if (player1.contains(1) && player1.contains(2) && player1.contains(3) || player1.contains(4) && player1.contains(
                5
            ) && player1.contains(6)
            || player1.contains(7) && player1.contains(8) && player1.contains(9) || player1.contains(
                1
            ) && player1.contains(4) && player1.contains(7)
            || player1.contains(2) && player1.contains(5) && player1.contains(8) || player1.contains(
                3
            ) && player1.contains(6) && player1.contains(9)
            || player1.contains(1) && player1.contains(5) && player1.contains(9) || player1.contains(
                3
            ) && player1.contains(5) && player1.contains(7)
        )
            winner = 1


        if (player2.contains(1) && player2.contains(2) && player2.contains(3) || player2.contains(4) && player2.contains(
                5
            ) && player2.contains(6)
            || player2.contains(7) && player2.contains(8) && player2.contains(9) || player2.contains(
                1
            ) && player2.contains(4) && player2.contains(7)
            || player2.contains(2) && player2.contains(5) && player2.contains(8) || player2.contains(
                3
            ) && player2.contains(6) && player2.contains(9)
            || player2.contains(1) && player2.contains(5) && player2.contains(9) || player2.contains(
                3
            ) && player2.contains(5) && player2.contains(7)
        )
            winner = 2
        if (winner != -1) {
            if (winner == 1) {
                Toast.makeText(this, "Winner is " + "Player 1", Toast.LENGTH_SHORT).show()
//                reset()

            } else {
                Toast.makeText(this, "Winner is Player 2", Toast.LENGTH_SHORT).show()
//                reset()
            }
        }

    }

//    private fun reset() {
//        button_1.isEnabled = true
//        button_1.text = ""
//        button_1.setBackgroundColor(Color.LTGRAY)
//
//        button_2.isEnabled = true
//        button_2.text = ""
//        button_2.setBackgroundColor(Color.LTGRAY)
//
//        button_3.isEnabled = true
//        button_3.text = ""
//        button_3.setBackgroundColor(Color.LTGRAY)
//
//        button_4.isEnabled = true
//        button_4.text = ""
//        button_4.setBackgroundColor(Color.LTGRAY)
//
//        button_5.isEnabled = true
//        button_5.text = ""
//        button_5.setBackgroundColor(Color.LTGRAY)
//
//        button_6.isEnabled = true
//        button_6.text = ""
//        button_6.setBackgroundColor(Color.LTGRAY)
//
//        button_7.isEnabled = true
//        button_7.text = ""
//        button_7.setBackgroundColor(Color.LTGRAY)
//
//        button_8.isEnabled = true
//        button_8.text = ""
//        button_8.setBackgroundColor(Color.LTGRAY)
//
//        button_9.isEnabled = true
//        button_9.text = ""
//        button_9.setBackgroundColor(Color.LTGRAY)
//
//    }

}